addons/sale/views/sale_order_views.xml

```diff
@@ -876,10 +876,11 @@
                                 <field name="invoice_status" invisible="1" groups="!base.group_no_one"/>
                             </group>
                             <group name="sale_shipping" string="Shipping">
-                                <label for="commitment_date" string="Delivery Date"/>
+                                <label for="commitment_date" string="Delivery Date" invisible="state != 'draft'"/>
+                                <label for="commitment_date" string="Promised Delivery" invisible="state =='draft'"/>
                                 <div name="commitment_date_div" class="o_row">
                                     <field name="commitment_date" readonly="state == 'cancel' or locked"/>
-                                    <span name="expected_date_span" invisible="not expected_date" class="text-muted">Expected: <field name="expected_date" class="oe_inline"/></span>
+                                    <span name="expected_date_span" class="text-muted" invisible="1"></span>
```

Comment:
```suggestion
```
What's the point on keeping a `<span>` that will never be displayed? :sweat_smile: